### ISSUE TITLE
Cannot enter text in input fields inside Carousel

### ISSUE THREAD

**Issue by mdazy** [FormidableLabs/nuka-carousel#694]
# Bugs and Questions

### Describe Your Environment

- What version of nuka-carousel are you using?
nuka-carousel@4.6.7
- What version of React are you using?
react@16.12.0
- What browser are you using?
Firefox 76.0.1, 

### Describe the Problem

See https://codesandbox.io/s/distracted-jones-5gofk

**Expected behavior:** The user should be able to input text in the input fields inside the Carousel children.
**Actual behavior:** It is not possible to input anything in the input fields inside the Carousel children - checked with both MUI's TextField and plain HTML input element.
It looks like the elements lose input focus very shortly after being clicked - if you type fast enough, after clicking to get focus you may manage to enter one character before focus gets lost.

### Additional Information
I have tried setting `dragging` and `swiping` props to false, but the problem persists.

I have checked older versions in CodeSandbox and it seems the last version with a correct behavior was 4.6.1.

However, inside my target application and in Storybook, 4.6.1 and 4.5.13 also exhibit the problematic behavior. The last version that appears to have the correct behavior in all contexts is 4.5.12.

**Comment by sarmeyer** [comment]
Hi @mdazy I've been looking into a way to solve this problem for a bit now. There is an extra `event.preventDefault` on the click handler because of some behavior in Safari that was causing the slide image to be dragged unnecessarily. If we remove that `preventDefault`, the input issue is solved, but then the behavior in Safari comes back. I'm trying to figure out the best solution for both bugs! 

If you're interested or want to take a pass at solving it, the offending code is in the `onMouseDown` method on `src/index.js`.

**Comment by unicornAppStartup** [comment]
> 
> 
> Hi @mdazy I've been looking into a way to solve this problem for a bit now. There is an extra `event.preventDefault` on the click handler because of some behavior in Safari that was causing the slide image to be dragged unnecessarily. If we remove that `preventDefault`, the input issue is solved, but then the behavior in Safari comes back. I'm trying to figure out the best solution for both bugs!
> 
> If you're interested or want to take a pass at solving it, the offending code is in the `onMouseDown` method on `src/index.js`.

For me that not work. I deleted event.preventDefault from index.js and nothing happend, but i **SOLVED** this by deleting:

` if (e && e.currentTarget) {
    e.currentTarget.focus();
  }`

in the **utilities/utilities.js** file in **handleSelfFocus** function and now text editing in input works fine. 
I set also dragging and swipeing to false. I know that probably isn't good solution by for now is ok.

Sorry for my English if something is wrong, but i'm 16 years old kid from Poland xd

**Comment by sarmeyer** [comment]
@unicornAppStartup - thanks for submitting that workaround! I am still working on a more permanent solution to the issue in the carousel itself.

**Comment by taikongfeizhu** [comment]
> > Hi @mdazy I've been looking into a way to solve this problem for a bit now. There is an extra `event.preventDefault` on the click handler because of some behavior in Safari that was causing the slide image to be dragged unnecessarily. If we remove that `preventDefault`, the input issue is solved, but then the behavior in Safari comes back. I'm trying to figure out the best solution for both bugs!
> > If you're interested or want to take a pass at solving it, the offending code is in the `onMouseDown` method on `src/index.js`.
> 
> For me that not work. I deleted event.preventDefault from index.js and nothing happend, but i **SOLVED** this by deleting:
> 
> ` if (e && e.currentTarget) { e.currentTarget.focus(); }`
> 
> in the **utilities/utilities.js** file in **handleSelfFocus** function and now text editing in input works fine.
> I set also dragging and swipeing to false. I know that probably isn't good solution by for now is ok.
> 
> Sorry for my English if something is wrong, but i'm 16 years old kid from Poland xd

it my change the event conditionin the utilities/utilities.js file in **handleSelfFocus** function at  like this:
```js
export var handleSelfFocus = function handleSelfFocus(e) {
  if (e && e.currentTarget) {
    if (e.target.nodeName !== 'INPUT'){
      e.currentTarget.focus();
    }
  }
};

```

**Comment by ayarmak** [comment]
Hi Team! Any update on this issue? FWIW, only dropping BOTH event.preventDefault AND e.currentTarget.focus() solves it for me, but I wonder if there are any plans to solve it within the package.

**Comment by ValGeorgiev** [comment]
Hey all, sorry for the short delay 😄 but the fix is coming #878 and it will be released soon. Thank you for your patience.

**Comment by ValGeorgiev** [comment]
This issue should be resolved in v5.0.10
